src/rules/unnecessaryElseRule.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * @license
+ * Copyright 2019 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as utils from "tsutils";
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+
+import { codeExamples } from "./code-examples/unnecessaryElse.examples";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        description: Lint.Utils.dedent`
+        Disallows \`else\` blocks following \`if\` blocks ending with a \`break\`, \`continue\`, \`return\`, or \`throw\` statement.`,
+        descriptionDetails: "",
+        optionExamples: [true],
+        options: null,
+        optionsDescription: "Not configurable.",
+        rationale: Lint.Utils.dedent`
+        When an \`if\` block is guaranteed to exit control flow when entered,
+        it is unnecessary to add an \`else\` statement.
+        The contents that would be in the \`else\` block can be placed after the end of the \`if\` block.`,
+        ruleName: "unnecessary-else",
+        type: "functionality",
+        typescriptOnly: false,
+        codeExamples,
+    };
+    /* tslint:disable:object-literal-sort-keys */
+
+    public static FAILURE_STRING(name: string): string {
+        return `The preceding \`if\` block ends with a \`${name}\` statement. This \`else\` block is unnecessary.`;
```

Comment:
Sorry to nitpick on this again: I think it might be a little more clear if the error message said <code>... This \`else\` is unnecessary</code> _(removing the `block`)_. Technically, the block's contents are fine; it's just the `else` that's unnecessary. Would you mind changing it?

Reply:
Done https://github.com/palantir/tslint/pull/4502/commits/f36d4201e2fa8ec3aec60a022db2bf51435aceaf